### DISCUSSION TITLE
Feedback: Stateless issues, making stuff up, ignoring details in favour of generic answers

### DISCUSSION THREAD

**Discussion by JonathanDotCel** [community/community#76347] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I asked Copilot chat to review my 1-page python script.
It tells me I am missing a default return. I am not.

I paste the snippet with the return in it, and say "wtf is this then?"
```
    #
    # Else, return the default line
    #
    return line
 ```

It says 
``` apologize for the confusion, but as an AI, I don't have the ability to retain or recall past interactions. Each interaction is stateless and doesn't have access to previous conversations or code snippets unless they are included in the same conversation turn```

I explain that it has just read my entire program and should know.
It claims it cannot.

Half the utility of something like ChatGPT is in how you can say "okay, but now do that in camel case", etc.
And ChatGPT is free.

So I ask it where I can leave feedback, it replies:
`https://github.com/github/copilot/discussions`
Not a real link

I start asking questions on compression and it's confidently giving me answers that only barely conform to what I'm asking.
It's missing important details such as compressing key value pairs individually, but with the same dictionary.
But no, I just get the generic zlib answer over and over till I get angry with it and tell it to pay attention.

In the end I have to tell it, if you don't know the answer, then just be quiet and tell me, don't waste my time.
I feel like maybe this is a better default stance than e.g. for chat apps.

Really looking forward to the future, but for now, it would be amazing if stuff like this could be adressed.

Thanks,
J

**Comment by outgaze** [comment]
I agree, it keeps repeating the same stuff as well a lot of the times and suggesting the same thing over and over like it's having sparkles i feel like it has more braindamage than the toxic gamers but even that would be a compliment for github copilot bahahaha

P.S: found out it's still using gpt 3.5 turbo for code generation and 4 for reasoning therefore we aren't really getting any improvements and it's not even the newest gpt 4 turbo it's still stuck in 2021 with old outdated and very hallucinating bad coding practices (although i somehow have managed to force it to use gpt 4 but it's using some knock off version of it or a really decapitated one) but eh that's what 10$ get you ig
